docarray/doc_index/abstract_doc_index.py

```diff
@@ -85,10 +86,13 @@ def __init__(self, db_config=None, **kwargs):
                 'A DocumentIndex must be typed with a Document type.'
                 'To do so, use the syntax: DocumentIndex[DocumentType]'
             )
+        self._logger = logging.getLogger('docarray')
```

Comment:
shouldn't we import the logger from where it is initially set up instead of calling `getLogger()` again? Maybe it's just me but that would feel nicer to me

Reply:
importing increase the risk of cyclic imports though. `getLogger()` would not? I am not sure about the general guidelines around this.

Reply:
maybe @girishc13 you have an opinion their? Otherwise we can stick with sentry blog reco

Reply:
> I see [sentry blog](https://blog.sentry.io/2022/07/19/logging-in-python-a-developers-guide/#python-logging-best-practices) recommending this: logging.getLogger(__name__)

I've also only encountered this approach, however, in that example, they create separate loggers for each module (i.e. they're using `getlogger` to create a new logger and not to get existing one), which is not similar to our case because currently we only have 1 logger ('docarray'). 

But this makes me wonder if we should initialize this one logger in the abstract docindex class and not in init.py since we only use it for docindex stuff. Thus, we won't have to call `getlogger` to get an existing logger.

Reply:
Sentry is a tool used for central server side logging. It makes sense to use `logging.getLogger(__name__)` when running a server with a specific log level and configuration for the whole application. This type of logging is a global control and reduces the flexibility. The main reason for using module level logging is to zoom in on specific functionalities of the docarray library.

I also think that we should not force the import and just use `logging.getLogger('docarray')`. The users are of course also allowed to import the logger.